Source/WebCore/WebCoreSwift/CryptoKitShim.swift

```diff
@@ -0,0 +1,94 @@
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+* THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import Foundation
+import CryptoKit
+
+@objc(WKAesGcmErrorCodes)
+public enum AESGCMErrorCodes: Int {
+    case Success = 0
+    case WrongTagSize = 1
+    case EncryptionFailed = 2
+    case EncryptionResultNil = 3
+    case InvalidArgument = 4
+    case TooBigArguments = 5
+}
+
+@objc(WKAesGcm)
+public final class AesGcm: NSObject {
+    
+    @objc public static func encrypt(
+        _ key: UnsafePointer<UInt8>,
+        _ keySize: UInt,
+        _ iv: UnsafePointer<UInt8>,
+        _ ivSize: UInt,
+        _ additionalData: UnsafePointer<UInt8>,
+        _ additionalDataSize: UInt,
+        _ plaintext: UnsafePointer<UInt8>,
+        _ plaintextSize: UInt,
+        _ ciphertext: UnsafeMutablePointer<UInt8>) -> Int
+    {
+        do {
+            if keySize > Int.max
+                || ivSize > Int.max 
+                || plaintextSize > Int.max
+                || additionalDataSize > Int.max {
+                return AESGCMErrorCodes.TooBigArguments.rawValue
+            }
+            if ivSize == 0 {
+                    return AESGCMErrorCodes.InvalidArgument.rawValue
+            }
+            let zeroArray = [UInt8](repeating: 0, count: 0);
+            let nonce = try AES.GCM.Nonce(data: UnsafeBufferPointer(start: iv, count: Int(ivSize)))
+            var message = UnsafeBufferPointer(start: plaintext, count: Int(plaintextSize))
+            if plaintextSize == 0 {
+                zeroArray.withUnsafeBufferPointer { ptr in 
+                    message = ptr
+                }
+            }
+            let symKey = SymmetricKey(data: UnsafeBufferPointer(start: key, count: Int(keySize)))
+            let sealedBox: AES.GCM.SealedBox
+            if additionalDataSize > 0 {
+                sealedBox = try AES.GCM.seal(
+                    message,
+                    using: symKey, 
+                    nonce: nonce,
+                    authenticating: UnsafeBufferPointer(start: additionalData, count: Int(additionalDataSize))
+                )
+            } else {
+                sealedBox = try AES.GCM.seal(
+                    message,
+                    using: symKey, 
+                    nonce: nonce
+                )
+            }
+            sealedBox.ciphertext.copyBytes(to: ciphertext, count: Int(plaintextSize));
+            sealedBox.tag.copyBytes(to: ciphertext + Int(plaintextSize), count: sealedBox.tag.count);
+        } catch {
+            return AESGCMErrorCodes.EncryptionFailed.rawValue
+        }
+        return AESGCMErrorCodes.Success.rawValue
```

Comment:
Echoing Alex, I think it should be fine to return enum cases here, since the `@objc` annotation on the enum itself will translate into an NS_ENUM.

Reply:
done.